enterprise/backend/test/metabase_enterprise/api/gsheets_test.clj

```diff
@@ -0,0 +1,211 @@
+                   :id "049f3007-2146-4083-be38-f160c526aca7",
+                   :created-at "2025-01-27T18:43:02Z"}]}]))
+
+(defn mock-make-request
+  ([responses method url] (mock-make-request responses method url nil))
+  ([responses method url body]
+   (get responses {:method method :url url :body body}
+        [:error {:status 404}])))
+
+(deftest gsheets-calls-pass-when-activated-and-superuser
+  (mt/with-premium-features #{:etl-connections :attached-dwh :hosting}
+    (mt/with-temporary-setting-values [api-key "non-blank"]
+      (with-redefs [hm.client/make-request (partial mock-make-request happy-responses)]
+        (is (malli=
+             [:map [:email [:maybe :string]]]
+             (mt/user-http-request :crowberto :get 200 "ee/gsheets/service-account")))))))
+
+(defn- ->zdt
+  [^long date ^long time ^String zone]
+  (ZonedDateTime/of (LocalDate/of date 1 1) (-> LocalTime/MIDNIGHT (.plusSeconds time)) (ZoneId/of zone)))
+
+(deftest sync-complete?-test
+  (let [earlier-time (->zdt 2000 0 "UTC")
+        later-time (->zdt 2022 0 "UTC")]
+
+    (is (not (#'gsheets.api/sync-complete? {:status "initializing" :last-dwh-sync nil :last-gdrive-conn-sync nil}))
+        "status must be active for sync to be complete")
+
+    (is (not (#'gsheets.api/sync-complete? {:status "active" :last-dwh-sync nil :last-gdrive-conn-sync nil}))
+        "sync is not complete when we don't get a last-gdrive-conn-sync time")
+
+    (is (not (#'gsheets.api/sync-complete? {:status "active" :last-dwh-sync nil :last-gdrive-conn-sync earlier-time}))
+        "sync is not complete when we don't get a last-dwh-sync time")
+
+    (is (not (#'gsheets.api/sync-complete? {:status "active" :last-dwh-sync later-time :last-gdrive-conn-sync nil}))
+        "sync is not complete when we don't get a last-gdrive-conn-sync time")
+
+    (is (not (#'gsheets.api/sync-complete? {:status "active" :last-dwh-sync earlier-time :last-gdrive-conn-sync later-time}))
+        "sync is not complete when the last dwh sync is before the last gdrive conn sync")
+
+    (is (not (#'gsheets.api/sync-complete? {:status "active" :last-dwh-sync later-time :last-gdrive-conn-sync later-time}))
+        "sync is not complete when the last dwh sync == the last gdrive conn sync")
+
+    (is (#'gsheets.api/sync-complete? {:status "active" :last-dwh-sync later-time :last-gdrive-conn-sync earlier-time})
+        "sync is complete when we get active status and the last local sync time is before current time")))
+
+(deftest can-get-service-account-test
+  (let [[status response] (mock-make-request happy-responses
+                                             :get
+                                             "/api/v2/mb/connections-google/service-account")]
+    (is (= :ok status))
+    (is (malli= [:map [:email [:string {:min 1}]]] (:body response)))))
+
+(def ^:private
+  gdrive-link
+  "nb: if you change this, change it in test_resources/gsheets/mock_hm_responses.edn"
+  "<gdrive-link>")
```

Comment:
is this supposed to be an actual link?

Reply:
oh, I guess it's just the value in the mocked response? maybe calling it `expected-gdrive-link` or something like that would be clearer